### PR TITLE
OSDOCS#9533.1: Updates for preflight cluster upgrades for ROSA Classic (peer review completed through PR: #76632)

### PR DESCRIPTION
**Pull request by fmcdonal**
Version(s):
4.16+

Issue:
https://issues.redhat.com/browse/OSDOCS-9533

Link to docs preview:
https://78825--ocpdocs-pr.netlify.app/openshift-rosa/latest/upgrading/rosa-upgrading-sts.html

QE review:
- [x] QE has approved this change.

Peer review completed through PR: https://github.com/openshift/openshift-docs/pull/76632

### CODE REVIEW

File: modules/rosa-how-upgrades-work.adoc
Comment (jameszwang) [inline modules/rosa-how-upgrades-work.adoc:26]: It would be better to use 'rosa describe upgrade --cluster=<cluster name|cluster_id>' instead of 'oc get clusterversion'

File: modules/rosa-upgrading-cli-tutorial.adoc
Comment (bchandra-ocp) [inline modules/rosa-upgrading-cli-tutorial.adoc:62]: " If the policies are not compatible with the chosen upgrade version, the CLI prompts you to upgrade them. "
Is this correct? Does CLI prompts users to upgrade them?

Comment (bchandra-ocp) [reply modules/rosa-upgrading-cli-tutorial.adoc:62]: What about UI? This paragraph is under the heading " Upgrading a ROSA Cluster" which I suppose is "generic" because there are topics separately for "Upgrading through ROSA CLI" and "Upgrading through OCM Console" that would be specific to CLI and console respectively?

File: modules/rosa-how-upgrades-work.adoc
Comment (bchandra-ocp) [inline modules/rosa-how-upgrades-work.adoc:42]: @ravitri Is this correct? I thought temporary worker node is ALWAYS created. It is just that it is not created in workers machine pool because customer can delete it.

File: upgrading/rosa-upgrading-sts.adoc
Comment (bchandra-ocp) [inline upgrading/rosa-upgrading-sts.adoc:32]: Again here, the CLI doesn't prompt users to upgrade them.  https://docs.google.com/document/d/1SQPZNiuWPP7_vQkp6wLR1_Wz45yFc718UQkq_qxRSwU/edit

**Review by bchandra-ocp — CHANGES_REQUESTED** [review]
We need to address some technical questions and incorrectness. Please refer to the shared documentation and sync up with previous writer for this EPIC.

File: upgrading/rosa-upgrading-sts.adoc
Comment (fmcdonal) [reply upgrading/rosa-upgrading-sts.adoc:13]: @bchandra-ocp confused a little.
Is it only line 13 that needs updating from:
`* Automated upgrades - Set an upgrade window for automatic recurring y-stream upgrades whenever a new version is available. Upgrades to minor versions must be manually scheduled`
to
`* Automated upgrades - Set an upgrade through the {cluster-manager} UI for automatic recurring y-stream upgrades whenever a new version is available. Upgrades to minor versions must be manually scheduled

File: modules/rosa-upgrading-cli-tutorial.adoc
Comment (fmcdonal) [reply modules/rosa-upgrading-cli-tutorial.adoc:62]: @ravitri can you comment on @bchandra-ocp questions: "Is this correct? Does CLI prompts users to upgrade them?" for end of line 62.

Comment (fmcdonal) [reply modules/rosa-upgrading-cli-tutorial.adoc:62]: @bchandra-ocp again I am confused as I was informed that this is for Classic only and like you said we have 2 separate modules/topics for CLI and Console - so does the OCM console not cover the UI? thanks

File: modules/rosa-how-upgrades-work.adoc
Comment (bchandra-ocp) [reply modules/rosa-how-upgrades-work.adoc:42]: What's documented is in fact correct. I stand corrected (because that was not my understanding). Now, I see this PR and it's documented correctly: https://github.com/openshift/managed-upgrade-operator/pull/382

Comment (bchandra-ocp) [reply modules/rosa-how-upgrades-work.adoc:42]: LGTM

File: upgrading/rosa-upgrading-sts.adoc
Comment (bchandra-ocp) [reply upgrading/rosa-upgrading-sts.adoc:32]: Please change this:
If the policies are not compatible with the chosen upgrade version, the CLI prompts you to upgrade them.
to 
If the policies are not compatible with the chosen upgrade version, the CLI automatically upgrades them.

File: modules/rosa-upgrading-cli-tutorial.adoc
Comment (fmcdonal) [reply modules/rosa-upgrading-cli-tutorial.adoc:62]: > " If the policies are not compatible with the chosen upgrade version, the CLI prompts you to upgrade them. " Is this correct? Does CLI prompts users to upgrade them?

Changed this to "automatically upgrades"

Comment (fmcdonal) [reply modules/rosa-upgrading-cli-tutorial.adoc:62]: > @bchandra-ocp again I am confused as I was informed that this is for Classic only and like you said we have 2 separate modules/topics for CLI and Console - so does the OCM console not cover the UI? thanks

@bchandra-ocp are you ok with this now, if you look at full preview:
https://76632--ocpdocs-pr.netlify.app/openshift-rosa/latest/upgrading/rosa-upgrading-sts

File: upgrading/rosa-upgrading-sts.adoc
Comment (bchandra-ocp) [inline upgrading/rosa-upgrading-sts.adoc:32]: If the policies are not compatible with the chosen upgrade version, the CLI automatically upgrades the cluster.

must be 

If the policies are not compatible with the chosen upgrade version, the CLI automatically upgrades IAM policies before upgrading the cluster.

**Review by bchandra-ocp — CHANGES_REQUESTED** [review]

**Comment by fmcdonal** [comment]
@bchandra-ocp all changes applied if you would like to see if I captured all:
https://78825--ocpdocs-pr.netlify.app/openshift-rosa/latest/upgrading/rosa-upgrading-sts

File: modules/rosa-upgrading-cli-tutorial.adoc
Comment (fmcdonal) [reply modules/rosa-upgrading-cli-tutorial.adoc]: @jameszwang so change from 
"
If you are upgrading an AWS Security Token Service (STS) cluster, this command starts an interactive IAM Roles/policies upgrade mode process that verifies the account and operator role policies for the chosen cluster are compatible with the target version of the upgrade. If the policies are not compatible with the chosen upgrade version, the CLI automatically upgrades them.
"
to?
do you want to reword for me so we get this right this time?

Comment (jameszwang) [reply modules/rosa-upgrading-cli-tutorial.adoc]: @fmcdonal I think it would be better like this:
“
If you are upgrading an AWS Security Token Service (STS) cluster, this command starts an interactive IAM Roles/policies upgrade mode process that verifies the account and operator role policies for the chosen cluster are compatible with the target version of the upgrade. If the policies are not compatible with the chosen upgrade version, the CLI automatically upgrades them in the auto mode.
“

Comment (fmcdonal) [reply modules/rosa-upgrading-cli-tutorial.adoc]: added "in auto mode", thanks for clarifying @jameszwang

**Comment by ravitri** [comment]
Thanks a lot everyone for your reviews and help. LGTM from my side as well :+1:

**Comment by fmcdonal** [comment]
I am removing the merge label they have just changed the go-live date to 28th

**Comment by kcarmichael08** [comment]
Hi @fmcdonal , when I looked at the merge review needed queue, the label was still there. But your last comment says it's not to go live until August 28, so should I not do the merge review yet?

File: modules/rosa-how-upgrades-work.adoc
Comment (kcarmichael08) [inline modules/rosa-how-upgrades-work.adoc:57]:
```suggestion
.. During the update of every node in the cluster, the cluster update waits until the time specified by the _node drain grace period_ to allow for safely draining the workloads. Upon reaching the node drain grace period, the node is forcibly drained to allow for cluster upgrade to progress. You can only configure the node drain grace period before initiating the upgrade and you cannot change it after the cluster upgrade begins.
```
https://www.ibm.com/docs/en/ibm-style?topic=word-usage#m

File: modules/rosa-how-upgrades-work.adoc
Comment (kcarmichael08) [inline modules/rosa-how-upgrades-work.adoc:56]:
```suggestion
.. During the update of each node, workloads are drained, honoring the PDBs. Workloads with PDBs that do not allow disruptions essentially block the draining of the node, increasing the elapsed time for the cluster update.
```
I wouldn't hold up merge for this, but maybe we can change it since there is another change in the file that I think should be made before merging (change "may").

File: modules/rosa-upgrading-manual-ocm.adoc
Comment (kcarmichael08) [inline modules/rosa-upgrading-manual-ocm.adoc:48]:
```suggestion
.. In the ROSA CLI, run the `rosa list operator-roles` command to list and verify that Operator roles associated with the cluster are compatible with the target minor version chosen for the upgrade. If not, run the `rosa upgrade operators-roles` command to upgrade the cluster's Operator roles to the latest OpenShift version.
```

File: upgrading/rosa-upgrading-sts.adoc
Comment (kcarmichael08) [inline upgrading/rosa-upgrading-sts.adoc:31]:
```suggestion
When a ROSA (classic architecture) cluster that uses AWS Security Token Services (STS) is upgraded, the ROSA CLI verifies the account and Operator role policies for the chosen cluster are compatible with the target version of the upgrade. If the policies are compatible, the CLI automatically upgrades the cluster. If the policies are not compatible with the chosen upgrade version, the CLI automatically upgrades IAM policies before upgrading the cluster. When scheduling the upgrade, you give administrative acknowledgment to confirm you have reviewed the changes involved with the upgrade, if required.
```

File: upgrading/rosa-upgrading-sts.adoc
Comment (kcarmichael08) [inline upgrading/rosa-upgrading-sts.adoc:19]: Hi, I guess this is a hard link maybe you can't use an xref since this is the ROSA doc set - but shouldn't it be linked to 4.16 since that is the latest version, if you have to use a hard link?

**Review by kcarmichael08 — COMMENTED** [review]
Hi, there are a couple of style issues that I think should be changed before merging. Also a question about an xref. Otherwise it looks good to merge, nice job!

File: upgrading/rosa-upgrading-sts.adoc
Comment (fmcdonal) [reply upgrading/rosa-upgrading-sts.adoc:19]: Changed to "You can use update channels to decide which Red Hat OpenShift Container Platform minor version to update your clusters to. {product-title} supports updates only through the `stable` channel. To learn more about OpenShift update channels and releases, see link:https://docs.openshift.com/container-platform/latest/updating/understanding_updates/understanding-update-channels-release.html[Understanding update channels and releases]."

**Pull request merged**
